tests/codegen_testcases/solidity/borsh_decoding_complex_types.sol

```diff
@@ -33,34 +33,28 @@ contract Testing {
 	    // CHECK: branch block3
         
         // CHECK: block5: # body
-	    // CHECK: ty:uint32 %1.cse_temp = (%temp.12 + uint32 4)
+	    // CHECK: ty:uint32 %temp.16 = (builtin ReadFromBuffer ((arg #0), %temp.12))
```

Comment:
Same as above, but now we are not validating the offset to read the `string` length.

Reply:
The validation here is for `offset + size_width + array_length`. I was talking about validating `offset + size_width`. As we don't have overflow detection turned on by default, `array_length` can be negative, so the former validation is going to be wrong.